src/Agent.Listener/Configuration/ConfigurationProvider.cs

```diff
@@ -150,50 +191,74 @@ public string GetServerUrl(CommandSettings command)
+        public string GetAgentWithSameNameAlreadyExistErrorString(AgentConfigSettings agentConfigSettings, string agentName)
         {
-            agent = await _agentServer.UpdateAgentAsync(poolId, agent);
-            await GetAndAddTags(agent, command);
+            return StringUtil.Loc("DeploymentMachineWithSameNameAlreadyExistInDeploymentGroup", ((DeploymentAgentConfigSettings)agentConfigSettings).DeploymentGroupId, agentName);
+        }
+
+        public async Task<TaskAgent> UpdateAgentAsync(AgentConfigSettings agentConfigSettings, TaskAgent agent, CommandSettings command)
+        {
+            var deploymentMachine = new DeploymentMachine() { Agent = agent };
+            deploymentMachine = await _deploymentGroupServer.ReplaceDeploymentMachineAsync(new Guid(((DeploymentAgentConfigSettings)agentConfigSettings).ProjectId), ((DeploymentAgentConfigSettings)agentConfigSettings).DeploymentGroupId, agent.Id, deploymentMachine);
 
-            return agent;
+            await GetAndAddTags(agentConfigSettings, agent, deploymentMachine.Id, command);
+            return deploymentMachine.Agent;
         }
 
-        public async Task<TaskAgent> AddAgentAsync(int poolId, TaskAgent agent, CommandSettings command)
+        public async Task<TaskAgent> AddAgentAsync(AgentConfigSettings agentConfigSettings, TaskAgent agent, CommandSettings command)
         {
-            agent = await _agentServer.AddAgentAsync(poolId, agent);
-            await GetAndAddTags(agent, command);
+            var deploymentMachine = new DeploymentMachine(){ Agent = agent };
+            deploymentMachine = await _deploymentGroupServer.AddDeploymentMachineAsync(new Guid(((DeploymentAgentConfigSettings)agentConfigSettings).ProjectId), ((DeploymentAgentConfigSettings)agentConfigSettings).DeploymentGroupId, deploymentMachine);
+            
+            await GetAndAddTags(agentConfigSettings, deploymentMachine.Agent, deploymentMachine.Id, command);
 
-            return agent;
+            return deploymentMachine.Agent;
         }
 
-        public Task DeleteAgentAsync(int agentPoolId, int agentId)
+        public async Task DeleteAgentAsync(AgentConfigSettings agentConfigSettings, AgentSettings settings, string currentAction)
         {
-            return _agentServer.DeleteAgentAsync(agentPoolId, agentId);
+            var machines = await GetDeploymentMachinesAsync(agentConfigSettings, settings.AgentName);
+            Trace.Verbose("Returns {0} machines with name {1}", machines.Count, settings.AgentName);
+            var machine = machines.FirstOrDefault();
+            if (machine == null)
+            {
+                _term.WriteLine(StringUtil.Loc("Skipping") + currentAction);
+            }
+            else
+            {
+                Guid projectGuid;
+                var isProjectIdAvailable = Guid.TryParseExact(((DeploymentAgentConfigSettings)agentConfigSettings).ProjectId, "D", out projectGuid);
```

Comment:
the name of the variable "isProjectIdAvailable" is confusing. Just add the Guid.TryParse.... directly in the if condition (). that is more readable